flake.nix

```diff
@@ -0,0 +1,51 @@
+{
+  inputs = {
+    nixpkgs.url = "github:serokell/nixpkgs";
+    haskell-nix.url = "github:input-output-hk/haskell.nix";
+    hackage = {
+      url = "github:input-output-hk/hackage.nix";
+      flake = false;
+    };
+    stackage = {
+      url = "github:input-output-hk/stackage.nix";
+      flake = false;
+    };
+    common-infra.url = "github:serokell/common-infra";
+  };
+
+  outputs = { self, nixpkgs, haskell-nix, hackage, stackage, common-infra }:
+    let
+      packagesFor = system: rec {
+        pkgs = nixpkgs.legacyPackages.${system}.extend
+          (nixpkgs.lib.foldl nixpkgs.lib.composeExtensions (_: _: { }) [
+            haskell-nix.overlay
+          ]);
+        backend = pkgs.callPackage ./backend { };
+      };
+    in {
+      packages = builtins.mapAttrs (system: _:
+        with packagesFor system; {
+          backend-lib = backend.library;
+          backend = backend.server // {
+            meta.modulePath = [ "services" "edna" "backend" ];
+          };
+        }) nixpkgs.legacyPackages;
+
+      ciSystems = [ "x86_64-linux" ];
+
+      checks = builtins.mapAttrs (system: _:
+        with packagesFor system; {
+          backend-test = backend.test;
+          backend-hlint = backend.hlint;
+        }) nixpkgs.legacyPackages;
+
+      pipelineFile = common-infra.mkPipelineFile self;
+
+      devShell = builtins.mapAttrs (system: packages:
+        let pkgs = nixpkgs.legacyPackages.${system};
+        in pkgs.mkShell {
+          inputsFrom = [ packages.backend-lib ];
+          buildInputs = with pkgs.haskellPackages; [ cabal-install hpack ];
```

Comment:
I don't use dev shell and I am not sure anybody needs it, but if we decided to keep it, probably we don't need `hpack` here because we don't use it (no `package.yaml`).

Reply:
`devShell` is the Flake equivalent of `shell.nix` -- and yes, I use it all the time.